### PR TITLE
feat(textfield): add textfield interactor

### PR DESCRIPTION
**Pull request by wKich**
- Extract the form field filters
- Add `TextField` interactor

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+export const createFormFieldFilters = <E extends HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>() => ({
+  valid: (element: E) => !element.labels?.[0].classList.contains("Mui-error"),
+  required: (element: E) => element.required,
+  description: (element: E) => {
+    const descriptionId = element.getAttribute("aria-describedby");
+    const descriptionElement = descriptionId ? element.ownerDocument.getElementById(descriptionId) : null;
+    return descriptionElement?.innerText ?? "";
+  },
+});
+
```
Comment (cowboyd) [inline src/helpers.ts]: Let's put this is a `form-field-filters.ts`

File: test/text-field.test.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        test
+          .step(textfield.focus())
+          .assertion(textfield.is({ focused: true }))
+      )
+  )
+  .child("required={true}", (test) =>
+    test
+      .step(renderTextField({ required: true }))
+      // NOTE Be aware this is not a regular `space` character
+      .assertion(TextField('textfield *').is({ required: true }))
```
Comment (cowboyd) [inline test/text-field.test.tsx]: Totally optional, but may want to use a regular expression here.
```suggestion
      .assertion(TextField(matching(/textfield\s+\*/)).is({ required: true }))
```

**Review by cowboyd — APPROVED** [review]

File: src/helpers.ts (hunk 1 above)
Comment (wKich) [reply src/helpers.ts]: Wait for merge #14

Comment (wKich) [reply src/helpers.ts]: Fixed

**Review by cowboyd — APPROVED** [review]

**Pull request merged**